### PR TITLE
[SPARK-18872][SQL][TESTS] New test cases for EXISTS subquery

### PR DESCRIPTION
**Pull request by dilipbiswal**
## What changes were proposed in this pull request?
This PR adds the first set of tests for EXISTS subquery.  

File name                        | Brief description
------------------------| -----------------
exists-basic.sql              |Tests EXISTS and NOT EXISTS subqueries with both correlated and local predicates.
exists-within-and-or.sql|Tests EXISTS and NOT EXISTS subqueries embedded in AND or OR expression.

DB2 results are attached here as reference :

[exists-basic-db2.txt](https://github.com/apache/spark/files/733031/exists-basic-db2.txt)
[exists-and-or-db2.txt](https://github.com/apache/spark/files/733030/exists-and-or-db2.txt)

## How was this patch tested?
This patch is adding tests.

### CODE REVIEW

**Comment by dilipbiswal** [comment]
cc @hvanhovell @gatorsmile @nsyca

File: sql/core/src/test/resources/sql-tests/results/subquery/exists-subquery/exists-basic.sql.out [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+                                  WHERE  dept.dept_id = emp.dept_id))
+-- !query 11 schema
+struct<emp_name:string,bonus_amt:double>
+-- !query 11 output
+emp 1	10.0
+emp 1	20.0
+emp 2	100.0
+emp 2	300.0
+emp 3	300.0
+emp 4	100.0
```
Comment (nsyca) [inline sql/core/src/test/resources/sql-tests/results/subquery/exists-subquery/exists-basic.sql.out]: I have compared this result file with the attached DB2 output. The two sets of result are equivalent.

File: sql/core/src/test/resources/sql-tests/results/subquery/exists-subquery/exists-within-and-or.sql.out [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+       SELECT emp_name 
+       FROM   emp 
+       WHERE  bonus_amt < emp.salary)
+-- !query 7 schema
+struct<emp_name:string,bonus_amt:double>
+-- !query 7 output
+emp 1	10.0
+emp 1	20.0
+emp 2	100.0
+emp 4	100.0
```
Comment (nsyca) [inline sql/core/src/test/resources/sql-tests/results/subquery/exists-subquery/exists-within-and-or.sql.out]: I have compared this result file with the attached DB2 output. The two sets of result are equivalent.

**Review by gatorsmile — APPROVED** [review]
I did the compare. The results are consistent with the output of DB2.

**Comment by gatorsmile** [comment]
LGTM

**Comment by gatorsmile** [comment]
Could you update the PR descriptions? Currently, it does not correctly describe the cases you added.

File: sql/core/src/test/resources/sql-tests/inputs/subquery/exists-subquery/exists-basic.sql [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+                           OR state = 'NJ');
+
+-- not exist both equal and greaterthan predicate
+-- TC.01.08
+SELECT * 
+FROM   bonus 
+WHERE  NOT EXISTS (SELECT * 
+                   FROM   emp 
+                   WHERE  emp.emp_name = emp_name 
+                          AND bonus_amt > emp.salary);
```
Comment (gatorsmile) [inline sql/core/src/test/resources/sql-tests/inputs/subquery/exists-subquery/exists-basic.sql]: Could you also add one more test case, which is very like `TC.01.08`? This test case is used very often in the real world. BTW, you do not need to run DB2. We can know the results using our naked eyes. : ) 
```SQL
-- select values present in one table but missing in another one

SELECT emp.*
FROM   emp
WHERE  NOT EXISTS (SELECT NULL
                   FROM   bonus
                   WHERE  bonus.emp_name = emp.emp_name);
```

Comment (dilipbiswal) [inline sql/core/src/test/resources/sql-tests/inputs/subquery/exists-subquery/exists-basic.sql]: @gatorsmile Thanks .. I have added the suggested test case.

**Comment by dilipbiswal** [comment]
retest this please

**Comment by gatorsmile** [comment]
Thanks! Merging to master.

**Comment by dilipbiswal** [comment]
Thank you very much @gatorsmile

**Pull request closed without merging**